src/client.ts

```diff
@@ -1339,17 +1343,35 @@ export class StreamChat<
     }
 
     /** update the client.state with any changes to users */
-    if (event.type === 'user.presence.changed' || event.type === 'user.updated') {
-      if (event.user?.id === this.userID) {
-        this.user = this.user && { ...this.user, ...event.user };
-        /** Updating only available properties in _user object. */
-        Object.keys(event.user).forEach((key) => {
-          if (this._user && key in this._user) {
-            /** @ts-expect-error */
-            this._user[key] = event.user[key];
-          }
-        });
+    if (
+      (event.type === 'user.presence.changed' || event.type === 'user.updated') &&
+      event.user.id === this.userID
+    ) {
+      // Remove deleted properties from user objects.
+      for (const key in this.user) {
+        if (key in event.user || isOwnUserBaseProperty(key)) {
+          continue;
+        }
+
+        if (this.user?.[key]) {
+          delete this.user[key];
+        }
+
+        if (this._user?.[key]) {
```

Comment:
whats the difference between user and _user here? just a question, not really a review feedback

Reply:
![Screenshot 2021-07-28 at 18 14 34](https://user-images.githubusercontent.com/11586388/127358557-56ed74d9-f58e-40de-a24d-cf9bf87722ad.png)

This comment might answer this one!!